### ISSUE TITLE
[Feature Request] Upgrade macOS release LLVM to 17

### ISSUE THREAD

**Issue by anyuzu99** [Ryujinx/Ryujinx#5811]
### Overview

Change to LLVM improves the performance, and take full use of new architecture.

### Smaller details

I tested by myself in my repo, the LLVM 17 works perfect here: https://github.com/anyuzu99/ryujinx-llvm-test/actions/runs/6542183238

### Nature of request

Change to LLVM improves the performance, and take full use of new architecture.

### Why would this feature be useful?

Change to LLVM improves the performance, and take full use of new architecture.

**Comment by gdkchan** [comment]
I don't think we can "update LLVM" since this is a .NET application, it is compiled using the C# and .NET JIT compiler. The runtime itself (at least the parts that are written in C++) is compiled using clang/LLVM on macOS, but we do not build the runtime ourselves. I think LLVM is also used on macOS packaging step, but it's just to create the app bundle, it shouldn't really affect performance.

**Comment by anyuzu99** [comment]
while I saw significant performance improvement between them, don't know why.

(the 1.1.4 is llvm 17 version)
<img width="1204" alt="Screenshot 2023-10-17 at 12 20 07" src="https://github.com/Ryujinx/Ryujinx/assets/16492117/05abc51a-23f1-4e8b-94ad-feffe4729246">
<img width="1204" alt="Screenshot 2023-10-17 at 12 22 51" src="https://github.com/Ryujinx/Ryujinx/assets/16492117/df54de9a-567e-4b86-910e-47b08ef0718a">

**Comment by IsaacMarovitz** [comment]
@jinyuzu99 This doesn't entirely make sense. Our usage of LLVM in the build pipleline is basically only for tool usage (lipo, otool, etc...) and it shouldn't be involved in the building process, and shouldn't have an effect on performance.
